apps/prairielearn/src/pages/instructorGradingJob/instructorGradingJob.js

```diff
@@ -12,20 +12,19 @@ const aws = require('../../lib/aws');
 const router = express.Router();
 const sql = sqldb.loadSqlEquiv(__filename);
 
-// NOTE: We assume that instructorGradingJob is only mounted on the course_instance
-// page route, as is currently the case. If you add a course page route, take care!
-// See instructorJobSequence, for example.
-
 router.get('/:job_id', (req, res, next) => {
   const params = {
     job_id: req.params.job_id,
-    course_instance_id: res.locals.course_instance.id,
+    course_instance_id: res.locals.course_instance?.id ?? null,
+    course_id: res.locals.course.id,
     authz_data: res.locals.authz_data,
     req_date: res.locals.req_date,
   };
-  sqldb.queryOneRow(sql.select_job, params, (err, result) => {
+  sqldb.queryZeroOrOneRow(sql.select_job, params, (err, result) => {
     if (ERR(err, next)) return;
-
+    if (result.rows.length === 0) {
+      return next(error.make(404, 'Job not found'));
+    }
```

Comment:
As discussed with Myles, I think 404ing if a job can't be found or isn't associated with the course in the URL is a better UX than 500ing because we didn't get the row we expected.